### PR TITLE
[Cellular Library] Fix build warning message.

### PR DESCRIPTION
**Pull request by andysun2015**
*Description of changes:*
- Fix build warning message.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/doxygen/pages.dox [hunk 1]
```diff
@@ -145,8 +145,8 @@ Some configuration settings are C pre-processor constants, and some are function
 @section CELLULAR_CONFIG_STATIC_ALLOCATION_CONTEXT
 @copydoc CELLULAR_CONFIG_STATIC_ALLOCATION_CONTEXT
 
-@section CELLULAR_CONFIG_STATIC_ALLOCATION_COMM_CONTEXT
```
Comment (chinglee-iot) [inline docs/doxygen/pages.dox:148]: Comm interface context is allocated in developer's porting now. Suggest to remove this config or we create another PR to review all the configs.

File: include/cellular_config_defaults.h [hunk 2]
```diff
@@ -331,8 +331,8 @@
  * <b>Possible values:</b>`0 or 1`<br>
  * <b>Default value (if undefined):</b> 0
  */
-#ifndef CELLULAR_CONFIG_STATIC_ALLOCATION_COMM_CONTEXT
```
Comment (chinglee-iot) [inline include/cellular_config_defaults.h:334]: Same as above reason.

**Review by chinglee-iot — APPROVED** [review]

File: docs/doxygen/pages.dox (hunk 1 above)
Comment (andysun2015) [reply docs/doxygen/pages.dox:148]: Currently this change is for misra C, I agree to create another PR for the purpose of removing.

File: include/cellular_config_defaults.h (hunk 2 above)
Comment (andysun2015) [reply include/cellular_config_defaults.h:334]: Currently this change is for the the warning from MISRA, I agree to create another PR for the purpose of removing.

File: test/unit-test/cellular_pktio_utest.c [hunk 3]
```diff
@@ -1650,7 +1647,7 @@ void test__Cellular_PktioSendAtCmd_Null_AtCmd( void )
     CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
     CellularContext_t context;
     CellularCommInterface_t * pCommIntf = &CellularCommInterface;
-    CellularCommInterfaceHandle_t commInterfaceHandle;
+    CellularCommInterfaceHandle_t commInterfaceHandle = ( CellularCommInterfaceHandle_t ) malloc( sizeof( CellularCommInterfaceHandle_t ) );
```
Comment (cobusve) [inline test/unit-test/cellular_pktio_utest.c:1650]: what happens when this malloc fails?

File: test/unit-test/cellular_pktio_utest.c [hunk 4]
```diff
@@ -1679,7 +1676,7 @@ void test__Cellular_PktioSendAtCmd_Invalid_String( void )
...
```
Comment (cobusve) [inline test/unit-test/cellular_pktio_utest.c:1679]: malloc fails?

File: test/unit-test/cellular_pktio_utest.c [hunk 5]
```diff
@@ -1707,7 +1704,7 @@ void test__Cellular_PktioSendAtCmd_Happy_Path( void )
...
```
Comment (cobusve) [inline test/unit-test/cellular_pktio_utest.c:1707]: malloc can fail

File: test/unit-test/cellular_pktio_utest.c [hunk 6]
```diff
@@ -1679,7 +1676,7 @@ void test__Cellular_PktioSendAtCmd_Invalid_String( void )
...
     CellularAtReq_t atReqSetRatPriority =
     {
         "AT+QCFG=\"Story for Littel Red Riding Hood: Once upon a time there was a dear little girl who was loved by every one who looked at her, but most of all by her grandmother, and there was nothing that she would not have given to the child. Once she gave her a little cap of red velvet, which suited her so well that she would never wear anything else. So she was always called Little Red Riding Hood.\"",
```
Comment (cobusve) [inline test/unit-test/cellular_pktio_utest.c:1682]: Spelling error

Comment (cobusve) [inline test/unit-test/cellular_pktio_utest.c:1682]: What is the purpose of this initialization?

File: test/unit-test/cellular_pktio_utest.c (hunk 3 above)
Comment (andysun2015) [reply test/unit-test/cellular_pktio_utest.c:1650]: That will cause error message after setting compile option -Werror
```
/mnt/c/Andy/project/cellular/lib/Lab-Project-FreeRTOS-Cellular-Library/test/unit-test/cellular_pktio_utest.c: In function ‘test__Cellular_PktioSendAtCmd_Null_AtCmd’:
/mnt/c/Andy/project/cellular/lib/Lab-Project-FreeRTOS-Cellular-Library/test/unit-test/cellular_pktio_utest.c:1664:28: error: ‘commInterfaceHandle’ is used uninitialized in this function [-Werror=uninitialized]
 1664 |     context.hPktioCommIntf = commInterfaceHandle;
      |     ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [unit-test/CMakeFiles/cellular_pktio_utest.dir/build.make:67: unit-test/CMakeFiles/cellular_pktio_utest.dir/cellular_pktio_utest.c.o] Error 1
```

File: test/unit-test/cellular_at_core_utest.c [hunk 7]
```diff
@@ -903,11 +903,10 @@ void test_Cellular_ATStrtoi_Happy_Path( void )
 {
     CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
     char * pStr = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_STRTOL_HAPPY_CASE_STRING ) + 1 ) );
```
Comment (cobusve) [inline test/unit-test/cellular_at_core_utest.c:905]: malloc

File: test/unit-test/cellular_at_core_utest.c [hunk 8]
```diff
@@ -1005,9 +1004,9 @@ void test_Cellular_ATIsPrefixPresent_Happy_Path( void )
 {
     CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
     char * pStr = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_PREFIX_STRING_INPUT ) + 1 ) );
```
Comment (cobusve) [inline test/unit-test/cellular_at_core_utest.c:1006]: Perhaps if you want it allocated and initialized:
char  pStr[CELLULAR_SAMPLE_PREFIX_STRING_INPUT] = {0};

Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:1006]: That sounds good, thanks for the suggestion.

File: test/unit-test/cellular_pktio_utest.c (hunk 3 above)
Comment (andysun2015) [reply test/unit-test/cellular_pktio_utest.c]: please refer to this [change](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/37/commits/d6ed9baaca207edc8cb5ab7fe890e2937975d2d8).

File: test/unit-test/cellular_at_core_utest.c (hunk 8 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c]: please refer to this [change](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/37/commits/d6ed9baaca207edc8cb5ab7fe890e2937975d2d8).

File: test/unit-test/cellular_at_core_utest.c (hunk 7 above)
Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c]: please refer to this [change](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/37/commits/d6ed9baaca207edc8cb5ab7fe890e2937975d2d8).

File: test/unit-test/cellular_pktio_utest.c (hunk 6 above)
Comment (andysun2015) [reply test/unit-test/cellular_pktio_utest.c:1682]: please refer to this [change](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/37/commits/d6ed9baaca207edc8cb5ab7fe890e2937975d2d8).

Comment (andysun2015) [reply test/unit-test/cellular_pktio_utest.c:1682]: please refer to this [change](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/37/commits/d6ed9baaca207edc8cb5ab7fe890e2937975d2d8).

File: test/unit-test/cellular_pktio_utest.c (hunk 4 above)
Comment (andysun2015) [reply test/unit-test/cellular_pktio_utest.c]: please refer to this [change](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/37/commits/d6ed9baaca207edc8cb5ab7fe890e2937975d2d8).

File: test/unit-test/cellular_pktio_utest.c (hunk 5 above)
Comment (andysun2015) [reply test/unit-test/cellular_pktio_utest.c]: please refer to this [change](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/37/commits/d6ed9baaca207edc8cb5ab7fe890e2937975d2d8).

File: test/unit-test/cellular_at_core_utest.c [hunk 9]
```diff
@@ -938,19 +934,17 @@ void test_Cellular_ATStrDup_Invalid_Param( void )
 {
     CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
     char * pDst;
-    char * pSrc;
+    char pSrc[ sizeof( CELLULAR_SAMPLE_PREFIX_STRING_LARGE_INPUT ) ] = { 0 };
```
Comment (cobusve) [inline test/unit-test/cellular_at_core_utest.c:937]: To be equivalent this would have to be CELLULAR_SAMPLE_PREFIX_STRING_LARGE_INPUT+1 ?

Comment (andysun2015) [reply test/unit-test/cellular_at_core_utest.c:937]: CELLULAR_SAMPLE_PREFIX_STRING_LARGE_INPUT is an character string array pre-defined in Macro.
The array of characters terminated by a null character '\0', and this character won't be counted by strlen() but sizeof().
So sizeof(arrary) equal to strlen(array) + 1.

File: common/src/cellular_pktio.c [hunk 10]
```diff
@@ -676,7 +676,7 @@ static CellularPktStatus_t _handleMsgType( CellularContext_t * pContext,
         if( *ppAtResp == NULL )
         {
             *ppAtResp = _Cellular_AtResponseNew();
-            LogDebug( ( "Allocat at response %p", *ppAtResp ) );
+            LogDebug( ( "Allocat at response %p", ( void * ) *ppAtResp ) );
```
Comment (cobusve) [inline common/src/cellular_pktio.c:679]: Spelling error

Comment (andysun2015) [reply common/src/cellular_pktio.c:679]: Thanks for the correction.

Comment (andysun2015) [reply common/src/cellular_pktio.c]: Please refer to [this change](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/pull/37/commits/86df4964329f419781572f4d629b65f6f5ec9a52) for the correction.

**Review by cobusve — APPROVED** [review]

**Review by chinglee-iot — APPROVED** [review]

**Pull request merged**